FirebaseDatabase/Sources/Realtime/FWebSocketConnection.m

```diff
@@ -100,40 +102,56 @@ - (instancetype)initWith:(FRepoInfo *)repoInfo
                                                      userAgent:userAgent
                                                    googleAppID:googleAppID
                                                  appCheckToken:appCheckToken];
-#if TARGET_OS_WATCH
-        // Regular NSURLSession websocket.
-        NSOperationQueue *opQueue = [[NSOperationQueue alloc] init];
-        opQueue.underlyingQueue = queue;
-        NSURLSession *session = [NSURLSession
-            sessionWithConfiguration:[NSURLSessionConfiguration
-                                         defaultSessionConfiguration]
-                            delegate:self
-                       delegateQueue:opQueue];
-        NSURLSessionWebSocketTask *task =
-            [session webSocketTaskWithRequest:req];
-        self.webSocketTask = task;
-
-        if (@available(watchOS 7.0, *)) {
-            [[NSNotificationCenter defaultCenter]
-                addObserverForName:WKApplicationWillResignActiveNotification
-                            object:nil
-                             queue:opQueue
-                        usingBlock:^(NSNotification *_Nonnull note) {
-                          FFLog(@"I-RDB083015",
-                                @"Received watchOS background notification, "
-                                @"closing web socket.");
-                          [self onClosed];
-                        }];
+
+        if (@available(iOS 13.0, macOS 10.15, macCatalyst 13.1, tvOS 13.0,
+                       watchOS 6.0, *)) {
+            // Regular NSURLSession websocket.
+            NSOperationQueue *opQueue = [[NSOperationQueue alloc] init];
+            opQueue.underlyingQueue = queue;
+            NSURLSession *session = [NSURLSession
+                sessionWithConfiguration:[NSURLSessionConfiguration
+                                             defaultSessionConfiguration]
+                                delegate:self
+                           delegateQueue:opQueue];
+            NSURLSessionWebSocketTask *task =
+                [session webSocketTaskWithRequest:req];
+            self.webSocketTask = task;
+
+#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_VISION || TARGET_OS_MACCATALYST
+            NSString *resignName = UIApplicationWillResignActiveNotification;
+#elif TARGET_OS_OSX
+            NSString *resignName = NSApplicationWillResignActiveNotification;
+#elif TARGET_OS_WATCH
+            NSString *resignName = WKApplicationWillResignActiveNotification;
+#elif
+#error("missing platform")
+#endif
+            if (@available(watchOS 7.0, *)) {
```

Comment:
Is this conditional's expression now outdated? What happens if running on watchOS 6?

Reply:
This is unchanged from before, since notifications aren't available on watchOS 6.